nasa/cFS-GroundSystem
# Resolves #103 Custom header mechanism

**Describe the contribution**
Fixes #103, implemented the ability to allow users to select the header version byte offset, or custom offsets. In order to support this, some relatively major changes had to be implemented:

- A Python-native "mini" `cmdUtil` to send UDP data. The GroundSystem app in its current form is no longer dependent on the external C `cmdUtil` app (also fixes #61, at least partially)
- Using Python's `mmap` module to share data between processes

**Testing performed**
Steps taken to test the contribution:
1. Make changes to code
2. Run program to verify functionality

**Expected behavior changes**
The update allows users to change the byte offsets for sending commands and parsing telemetry, to support different header versions or other implementations of cFS

**Additional context**
- Note that the "mini" `cmdUtil` is only a subset of `cmdUtil`'s full functionality. Should the GroundSystem app be updated in the future, the full `cmdUtil` might need to be used again, or the "mini" `cmdUtil` updated accordingly
- There is a larger issue regarding the GroundSystem app as a whole and interprocess communication (IPC). There are multiple techniques to achieve this, but which technique is preferred, and whether or not GroundSystem should be completely redesigned to avoid IPC in the first place, is a discussion for the CCB.


**Contributor Info - All information REQUIRED for consideration of pull request**
Leor Bleier, NASA GSFC\Code 582
